aws/code/MLWorkflow.py

```diff
@@ -0,0 +1,63 @@
+from abc import abstractmethod
+import functools
+import time
+
+class MLWorkflow ( ):
+    def __init__ ( self ):
+        return None
+
+    def create_workflow( self, hpo_config ):        
+        if hpo_config.compute_type == 'single-CPU':
+            from workflows.MLWorkflowSingleCPU import MLWorkflowSingleCPU
```

Comment:
Any particular reason for the delayed import here?

Reply:
Without the delay, it becomes a circular import (i.e. in the child class there is an import of the parent which is not fully defined yet). Maybe there is a better way to do this?

Reply:
Ah, of course! In that case, I'd pull this out as a separate factory function to break the circular dependency. I just noticed that my linter recommends that anyway, since this method makes no use `self` or `cls`.